File: php.local/www/experiment/compilation/literal/literal.php
<?php
// literal.php

$operator = $_GET['operator'] ?? 'divide';
try {
	if ($operator == 'divide'){
		$result = 1 / 0;
	}else{
		$result = 1 % 0;
	}
} catch (Throwable $t){
	echo "Operator: $operator<br>";
	printf("Type: %s<br>", get_class($t));
	printf("Message: %s<br>", $t->getMessage());
}

File: php.local/www/experiment/compilation/literal/literal.rb
## literal.rb

operator = ARGV[0] || "divide"
begin
	if operator == 'divide' then
		result = 1 / 0
	else
		result = 1 % 0
	end
rescue Exception => e
	puts "Operator: #{operator}"
	puts "Message: #{e.exception}"
	puts "Class: #{e.class}"
end
